File: py/search/linked_list/SequenceSearch.py
#!/usr/bin/env python
# -*- coding:gbk -*-

class ListNode:
    def __init__(self, value: int):
        self.value, self.next = value, None

class SequenceSearch:
    def search(self, head: ListNode, target: int) -> ListNode:
        current = head.next
        while current:
            if target == current.value:
                return current
            current = current.next
        return None

File: java/search/linked_list/SequenceSearch.java
package search.linked_list;

public class SequenceSearch {
    public ListNode search(ListNode head, int target) {
        ListNode current = head.next;
        while (null != current) {
            if (target == current.value) {
                return current;
            }
            current = current.next;
        }
        return null;
    }

    public class ListNode {
        public int value = 0;
        public ListNode next = null;

        public ListNode(int value) {
            this.value = value;
        }
    }
}
